packages/bitcore-node/src/providers/chain-state/evm/api/csp.ts

```diff
@@ -495,25 +495,33 @@ export class BaseEVMStateProvider extends InternalStateProvider implements IChai
           }
         }
 
+        let _value;
+        if (data) {
+          // Gas estimation might fail with `insufficient funds` if value is higher than balance for a normal send.
+          // We want this method to give a blind fee estimation, though, so we should not include the value
+          // unless it's needed for estimating smart contract execution.
+          _value = web3.utils.toHex(value)
+        }
+
         const opts = {
           method: 'eth_estimateGas',
           params: [
             {
               data,
               to: to && to.toLowerCase(),
               from: from && from.toLowerCase(),
-              gasPrice: web3.utils.toHex(gasPrice),
-              value: web3.utils.toHex(value)
+              // gasPrice: web3.utils.toHex(gasPrice), // Setting this lower than the baseFee of the last block will cause an error. Better to just leave it out.
+              value: _value
             }
           ],
           jsonrpc: '2.0',
-          id: 1
+          id: 'bitcore-' + Date.now()
         };
 
         let provider = web3.currentProvider as any;
         provider.send(opts, (err, data) => {
           if (err) return reject(err);
-          if (!data.result) return reject(data.message);
+          if (!data.result) return reject(data.error || data);
```

Comment:
tests need to reflect this change 
https://github.com/bitpay/bitcore/blob/master/packages/bitcore-node/test/unit/modules/ethereum/csp.spec.ts#L223